### ISSUE TITLE
Not 200 solr response cause crash on python3

### ISSUE THREAD

**Issue by touilleMan** [toastdriven/pysolr#159]
Hi there,

I've encountered an encoding error (see: https://github.com/toastdriven/pysolr/blob/master/pysolr.py#L445)

```
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/app/.heroku/python/lib/python3.4/site-packages/pysolr.py", line 813, in add
    return self._update(m, commit=commit, softCommit=softCommit, waitFlush=waitFlush, waitSearcher=waitSearcher)
  File "/app/.heroku/python/lib/python3.4/site-packages/pysolr.py", line 399, in _update
    return self._send_request('post', path, message, {'Content-type': 'text/xml; charset=utf-8'})
  File "/app/.heroku/python/lib/python3.4/site-packages/pysolr.py", line 328, in _send_request
    error_message = self._extract_error(resp)
  File "/app/.heroku/python/lib/python3.4/site-packages/pysolr.py", line 417, in _extract_error
    reason, full_html = self._scrape_response(resp.headers, resp.content)
  File "/app/.heroku/python/lib/python3.4/site-packages/pysolr.py", line 445, in _scrape_response
    if response.startswith('<?xml'):
TypeError: startswith first arg must be bytes or a tuple of bytes, not str
```

Basically, the issue is `response` (i.e. response's content) is made of bytes where `'<?xml'` is an unicode string on python3.
We could solve this pretty easily by prefixing the string with a `b` to indicate it is supposed to be a bytes string under python3 (and a regular str under python2)

Do you want me to provide this pretty simple fix ?

**Comment by DrChai** [comment]
It happens on Python 3.4.3 as all I known (i just tested on python 3.4.0 and 3.4.3, it works for v3.4.0).
Reason is the response variable is bytes not a string. you have to change the line 445 to:
response.startswith(b'<?xml'): // makes '<?xml' in bytes
and then call
sudo python setup.py install

**Comment by touilleMan** [comment]
@DrChai I think you should provide a (pretty simple) PR for this issue (but if you don't have time for this I can do it)
@toastdriven Do you agree about a PR ?

**Comment by DrChai** [comment]
@touilleMan you could just do PR for this. 
And i was wrong about its working on 3.4.0. I just realized it just didnt have a chance to call that function.
More details u can take a look at this answer:
http://stackoverflow.com/questions/6269765/what-does-the-b-character-do-in-front-of-a-string-literal/6269785#6269785

**Comment by dhbahr** [comment]
Maybe upstream could include support for both things, something like

            ...

            try:
                rc = response.startswith('<?xml')
            except TypeError:
                rc = response.startswith(b'<?xml')
            if rc:
                ...

**Comment by touilleMan** [comment]
@toastdriven Given this bug is pretty crittical for python3, do you have any plan to release a v3.3.3 with this fixed ?

**Comment by acdha** [comment]
I just released v3.3.3 on PyPI:

https://pypi.python.org/pypi/pysolr/3.3.3

**Comment by touilleMan** [comment]
awesome :+1:
